### PR TITLE
Fix JoinDetection when CrossJoin is someone's right child

### PR DESCRIPTION
**Pull request by mrzzzrm**
Since god only knows when we'll be able to discard the JoinDetection rule, let's keep it healthy

### CODE REVIEW

File: src/lib/optimizer/strategy/join_detection_rule.cpp [hunk 1]
```diff
@@ -98,8 +106,18 @@ std::optional<JoinDetectionRule::JoinCondition> JoinDetectionRule::_find_predica
...
+      auto predicate_right_column_id = boost::get<ColumnID>(predicate_node->value());
+
+      // The Predicate refers to a subtree left of the subtree the Cross Join came from
+      if (predicate_left_column_id < column_id_offset || predicate_right_column_id < column_id_offset) {
+        continue;
+      }
+
+      // Calculate column ids relative to the leftmost output column of the CrossJoin subtree.
+      predicate_left_column_id -= column_id_offset;
+      predicate_right_column_id -= column_id_offset;
```
Comment (mrks) [inline src/lib/optimizer/strategy/join_detection_rule.cpp]: Why can't these become negative here?

File: src/lib/optimizer/strategy/join_detection_rule.cpp [hunk 2]
```diff
@@ -98,8 +106,18 @@ std::optional<JoinDetectionRule::JoinCondition> JoinDetectionRule::_find_predica
...
        * we can do this by the simple range check in _is_join_condition()
        */
-      const auto predicate_left_column_id = predicate_node->column_id();
-      const auto predicate_right_column_id = boost::get<ColumnID>(predicate_node->value());
+      auto predicate_left_column_id = predicate_node->column_id();
+      auto predicate_right_column_id = boost::get<ColumnID>(predicate_node->value());
+
+      // The Predicate refers to a subtree left of the subtree the Cross Join came from
+      if (predicate_left_column_id < column_id_offset || predicate_right_column_id < column_id_offset) {
+        continue;
```
Comment (mrks) [inline src/lib/optimizer/strategy/join_detection_rule.cpp]: Why do we only have to check the left side here?

Comment (mrzzzrm) [reply src/lib/optimizer/strategy/join_detection_rule.cpp]: Right side check happens in `::_is_join_condition`

File: src/lib/optimizer/strategy/join_detection_rule.cpp (hunk 1 above)
Comment (mrks) [reply src/lib/optimizer/strategy/join_detection_rule.cpp]: Ah, my fault. I misunderstood the algorithm.

**Pull request merged**